tests/e2e/scenarios/test_reborn_qa_trace_full_path.py

```
@@ -220,43 +687,224 @@ async def _wait_for_trace_replay(mock_llm_server: str, timeout: float = 30) -> d
+                params=params,
+                timeout=15,
+            )
+            if response.status_code != 429:
+                response.raise_for_status()
+                page = response.json()
+                break
+            await asyncio.sleep(0.5)
+        else:
+            raise AssertionError(
+                "timeline remained rate-limited after replay completed"
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)